NitroxClient/MonoBehaviours/RemotePlayerColorApplicator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using NitroxClient.GameLogic;
+using NitroxClient.GameLogic.PlayerModel;
+using NitroxClient.GameLogic.PlayerModel.Abstract;
+using NitroxClient.GameLogic.PlayerModel.ColorSwap;
+using NitroxModel.Core;
+using UnityEngine;
+
+namespace NitroxClient.MonoBehaviours
+{
+    public class RemotePlayerColorApplicator : MonoBehaviour
+    {
+        private IEnumerable<IColorSwapManager> colorSwapManagers;
+        private RemotePlayer remotePlayer;
+        private ColorSwapAsyncOperation swapOperation;
+
+        public void Awake()
+        {
+            colorSwapManagers = NitroxServiceLocator.LocateService<IEnumerable<IColorSwapManager>>();
+        }
+
+        public void Update()
+        {
+            if (remotePlayer == null)
+            {
+                return;
+            }
+
+            if (swapOperation != null)
+            {
+                return;
+            }
+
+            swapOperation = new ColorSwapAsyncOperation(remotePlayer, colorSwapManagers);
+            StartCoroutine(ApplyPlayerColor());
+        }
+
+        public void AttachRemotePlayer(RemotePlayer remotePlayer)
+        {
+            if (this.remotePlayer != null)
+            {
+                throw new InvalidOperationException("An applicator instance may only be used once.");
+            }
+
+            this.remotePlayer = remotePlayer;
+        }
+
+        private IEnumerator ApplyPlayerColor()
+        {
+            swapOperation.BeginColorSwap();
+
+            while (!swapOperation.IsColorSwapComplete())
```

Comment:
In my research I found out that it was better for Unity if you use WaitForSeconds instead of WaitUntil if you knew that the operation was going to take more than just a few frames to complete. 

The reason being is that Unity will default to checking on the status of the job with each frame update. If the game is running at 30 fps and the operation takes 5 seconds, then the status of the job will be evaluated 150 times. That number can be reduced to 51 (at most) by telling Unity to check the status every 1/10th of a second instead.

I simply went with what was suggested on the internet given my complete lack of experience with this level of Unity development as well. If I am being honest: the number of checks seems like it should be irrelevant given the scale. I just don't know enough to be able to call BS.

Perhaps @nesrak1 would be a good person to ask?

Reply:
Yes you're right, wait until will check every frame for the condition. As long as it's not very intensive it's probably not a big deal. If it is, you might want to wait every .1 seconds.